src/Adapter/Customer/QueryHandler/GetRequiredFieldsForCustomerHandler.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\QueryHandler;
+
+use Customer;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Query\GetRequiredFieldsForCustomer;
+use PrestaShop\PrestaShop\Core\Domain\Customer\QueryHandler\GetRequiredFieldsForCustomerHandlerInterface;
+
+/**
+ * Handles query which gets required fields for customer sing up
+ *
+ * @internal
+ */
+final class GetRequiredFieldsForCustomerHandler implements GetRequiredFieldsForCustomerHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetRequiredFieldsForCustomer $query)
+    {
+        $requiredFields = (new Customer())->getFieldsRequiredDatabase();
+
+        if (empty($requiredFields)) {
+            return [];
+        }
+
+        $fields = [];
+
+        foreach ($requiredFields as $field) {
```

Comment:
`array_map()` ? Or is it less performing ?

Also please check for `field_name` array key to exist. I know that this is supposed to be ensured by database schema but if user does modify its database schema and then this page stops working, we wish he can easily find the mistake with an helpful error message 😉 . It can't hurt.

Reply:
> if user deletes table or even database, should we handle that as well

IMO yes: we should handle it by throwing an exception with an helpful message that will allow user to quickly understand what is wrong.

It's 3 lines of code for us (one `if`, one `throw Exception` and one closing brack `}`) but can really help the user. Do not use a domain exception (`RuntimeException` is fine here), do not use a translated and complex error message, use some short one. So it's a small thing for us, and a great help for the PrestaShop developer.

So:
```
    if (false === array_key_exists('field_name', $field)) {
        throw new \RuntimeException('Expects property field_name in $field, but it does not exists.');
    }
    $fields[] = $field['field_name'];
}
```

Dont forget PrestaShop is a opensource CMS: people can take it and start modifying it for their needs. So they might replace the mysql database with a nosql database or an API. We will not write all our code for this usecase 😛 but the php code should check whether the input it is given is right, and crash else. It's [defensive programming](https://en.wikipedia.org/wiki/Defensive_programming).

We can discuss it with @mickaelandrieu if you want 😉 maybe I'm too paranoid but when I look at PrestaShop bugs history I think it's relevant here.

Reply:
im aware of defensive programming but checking if fields exist from database seems like a big overhead. :/ 
however, i dont want this PR to be blocked by this "problem", could you create an issue in prestashop so we can discuss that as it would affect a lot of code.